### DISCUSSION TITLE
Code annotation for a custom language?

### DISCUSSION THREAD

**Discussion by hellt** [squidfunk/mkdocs-material#5017] [Q&A] [answered]
Hi @squidfunk 
I added a custom syntax highlighter to mkdocs-material, and I wonder if I can make code annotations to work with this custom lang?
How/where do I say which char mkdocs-material should consider a comment to make the following notation to work?

```
#(1)!
```

**Comment by squidfunk** [comment, marked as answer]
Use one of the following CSS classes:

https://github.com/squidfunk/mkdocs-material/blob/48e8b2be30fed05c76dda6bc20ff039ed3e8601f/src/assets/javascripts/components/content/annotation/list/index.ts#L74

**Reply by hellt** [reply]
ahh, damn, my pygments parser failed to mark `#()` as a comment element.
Gotcha, thanks, will fix that and retry.

**Comment by squidfunk** [comment]
With [Insiders 4.32.0](https://squidfunk.github.io/mkdocs-material/insiders/changelog/#4.32.0), code annotations can now be placed anywhere. Material for MkDocs can be configured per-language to allow for code annotations in other lexemes than comments, e.g. strings. See the documentation on [custom selectors](https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#custom-selectors).

**Reply by hellt** [reply]
thanks @squidfunk 
this is cool. It might be worthwhile adding a note, that for custom lexemes the code annotation syntax stays unchanged (except `!` is not needed)

e.g.

``` json
{
  "key": "value (1)"
}
```

**Reply by squidfunk** [reply]
Agree, I still need some time to rework the documentation a little. Wanted to push this out first. `!` may be useful at times, for example when you want to add inline bash annotations:

``` sh
docker "(1)!" run squidfunk/mkdocs-material
```

This would render as `docker (+) run squidfunk/mkdocs-material`, as opposed to `docker "(+)" run squidfunk/mkdocs-material` when the `!` would be omitted. Note that this breaks examples when JavaScript is not available (= too slow). We'll see how we can mitigate that in the future, but it should be rarely relevant. Printing works as well.
